src/lib.rs

```diff
@@ -753,17 +749,21 @@ impl<T: Counter> Histogram<T> {
     ///
     /// Returns an error if `value` exceeds the highest trackable value and auto-resize is
     /// disabled.
-    pub fn record_n_correct(&mut self, value: i64, count: T, interval: i64) -> Result<(), ()> {
+    pub fn record_n_correct(&mut self, value: u64, count: T, interval: u64) -> Result<(), ()> {
         try!(self.record_n(value, count));
         if interval <= 0 {
             return Ok(());
         }
 
-        let mut missingValue = value - interval;
-        while missingValue >= interval {
+        if value > interval {
```

Comment:
I wonder if this should actually be

```rust
let mut missingValue = value;
while missingValue >= interval {
    missingValue -= interval;
    try!(self.record_n(missingValue, count));
}
```

The semantics of that are slightly different (i.e., the "last" `interval` will also be counted), but I think that's how it is supposed to be (i.e., the old version was also wrong).